addons/link_tracker/tests/test_mail_render_mixin.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import common
+
+
+class TestMailRenderMixin(common.TransactionCase):
+    def test_shorten_links(self):
+        self.env["mail.render.mixin"]._shorten_links(
```

Comment:
There are some existing tools in common that you could re-use.

Side note: could you add a test with something like "https://www.my_url.com?foo=bar&baz=qx" ?

Reply:
@tde-banana-odoo Ok for the test, about common there's just function to parse the HTML, and here I check the link tracker creation